client/CPlayerInterface.cpp

```diff
@@ -198,48 +199,184 @@ void CPlayerInterface::playerStartsTurn(PlayerColor player)
+		if(playerInfo.isControlledByHuman())
+		{
+			playerNames += playerInfo.name + ", ";
+			humanPlayerCount++;
 		}
+	}
+	playerNames.pop_back();
+	playerNames.pop_back();
+	isMulitplayerGame = humanPlayerCount > 1;
+
+	// normalize map name
+	//int txtlen = TextOperations::getUnicodeCharactersCount(mapName); 
+	//TextOperations::trimRightUnicode(mapName, std::max(0, txtlen - 15)); // shorten to max 15 chars
+	std::string forbiddenChars("\\/<>:\"|?*\t\n\v\f\r"); 
+	std::replace_if(mapName.begin(), mapName.end(), [&](char c) { return std::string::npos != forbiddenChars.find(c); }, '_' ); // replace forbidden chars with _
+
+	// store variable values in map
+	std::unordered_map<std::string, std::string> variables;
+	variables["mapName"] = mapName;
+	variables["templateName"] = mapName;
+	variables["timestamp"] = timeStamp;
+	variables["turn"] = turn;
+	variables["players"] = playerNames;
+	variables["campaignName"] = campaignName;
+ 
+	if(simpleSaves)
+	{
+		int autosaveCountLimit = settings["general"]["autosaveCountLimit"].Integer(); 
+		// save using a counter, and overwrite depending on autosave count limit
+		if(autosaveCountLimit > 0) 
+		{
+			auto saveNumber = extractNumberAfterHash(lastMapSaveName);
+			if(saveNumber != -1)
+				autosaveCount = saveNumber;
 
-		autosaveCount++;
+			autosaveCount++;
 
-		int autosaveCountLimit = settings["general"]["autosaveCountLimit"].Integer();
-		if(autosaveCountLimit > 0)
+			saveName = "Autosave#" + std::to_string(autosaveCount);
+			autosaveCount %= autosaveCountLimit; 
+		}
+		// save using curent turn (EG: 113) and no limit
+		else 
+			saveName = "Autosave_" + turn; 
+	}
+	else
+	{
+		// determine save type
+		if(isMulitplayerGame)
 		{
-			cb->save("Saves/Autosave/" + prefix + std::to_string(autosaveCount));
-			autosaveCount %= autosaveCountLimit;
+			switch(mapType)
+			{
+			case RandomMap:
+				saveName = settings["general"]["multiPlayerRmgSaveName"].String(); break;
+			case Scenario:
+				saveName = settings["general"]["multiPlayerScenarioSaveName"].String(); break;
+			}
+			
+			// prepend [hotseat] before player names
+			if(isHotseat)
+			{
+				size_t insertPos = saveName.find("%players%");
+				if(insertPos != std::string::npos)
+					saveName.insert(insertPos, "[hotseat]");
+			}
 		}
 		else
 		{
-			std::string stringifiedDate = std::to_string(cb->getDate(Date::MONTH))
-					+ std::to_string(cb->getDate(Date::WEEK))
-					+ std::to_string(cb->getDate(Date::DAY_OF_WEEK));
-
-			cb->save("Saves/Autosave/" + prefix + stringifiedDate);
+			switch(mapType)
+			{
+			case RandomMap:
+				saveName = settings["general"]["singlePlayerRmgSaveName"].String(); break;
+			case Scenario:
+				saveName = settings["general"]["singlePlayerScenarioSaveName"].String(); break;
+			case Campaign :
+				saveName = settings["general"]["singlePlayerCampaignSaveName"].String(); break;
+			}
 		}
-	}
+
+		saveName = replaceVariables(saveName, variables);

```

Comment:
You can use `boost::replace_all` instead of this method, e.g.
```cpp
for (auto const & entry : variables)
    boost::replace_all(saveName, "%" + entry.first + "%", entry.second);
```

Reply:
is boost not shunned everywhere in favor of std:: ?

Reply:
If there is *identical* functionality in std - maybe.
But here boost provides way more convenient text formatting tools

Reply:
Done